src/Microsoft.OData.Client/DataServiceQueryOfT.cs

```diff
@@ -314,14 +314,27 @@ public DataServiceQuery<TElement> Expand<TTarget>(Expression<Func<TElement, TTar
         /// <returns>A new <see cref="T:Microsoft.OData.Client.DataServiceQuery`1" /> object that has the inline count option set.</returns>
         public DataServiceQuery<TElement> IncludeTotalCount()
         {
-            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethod("IncludeTotalCount");
+            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethods()
+                .FirstOrDefault(m => m.Name == "IncludeTotalCount" && m.GetParameters().Length == 0);

```

Comment:
After opting for ```.FirstOrDefault``` you would need to take care of the ```OrDefault``` part of the result, what if ```mi == null```?

Reply:
+1

---
In reply to: [375320794](https://github.com/OData/odata.net/pull/1658#discussion_r375320794) [](ancestors = 375320794)

Reply:
Switched to using `First`

Reply:
@KenitoInc That will just throw an exception if there's no "First". Might be better to just stick to `FirstOrDefault `and handle `mi == null`